atef/widgets/config/utils.py

```diff
@@ -1402,6 +1405,68 @@ def set_widget_font_size(widget: QWidget, size: int):
     widget.setFont(font)
 
 
+_float_re = re.compile(r'(([+-]?\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?)')
+
+
+def valid_float_string(string):
+    match = _float_re.search(string)
+    return match.groups()[0] == string if match else False
+
+
+class FloatValidator(QValidator):
```

Comment:
There are a couple of other `QValidator` signals/methods that we need but don't override.  It's not just the validate and fixup methods.

Reply:
There's nothing like `setValidator` on `QDoubleSpinBox`, is there? 

I'm confused how the instance is used beyond the two bound methods implemented above (which don't reference `self` even? Hmm)

Reply:
Oh I see what you mean.  I had assumed the `validator` attribute on `QSpinBox` was an actual validator.  This can be removed and lumped into the spinbox itself